src/core/chart-transformers.js

```diff
@@ -0,0 +1,283 @@
+    throw new Error("Cannot build variable response chart from single variable chart");
+  }
+  series = series[0];
+
+  //see if this series has an explicit axis association, default to y if not.
+  const axis = graph.data.axes[series] ? graph.data.axes[series] : 'y';
+
+  return _.isString(graph.axis[axis].label) ?
+      graph.axis[axis].label :
+      graph.axis[axis].label.text;
+};
+
+/***************************************************************************
+ * 1. makeAnomalyGraph and its helper functions
+ ***************************************************************************/
+//TODO: this function needs a test
+/*
+ * Graph transformation function that accepts a graph containing one or more
+ * timeseries associated with a single axis, and the name of one of the 
+ * timeseries to use as a base. Adds a secondary y axis, graphing a data series
+ * showing the difference (anomaly) between each of the original series and the
+ * base series. (Including the base series itself, shown as a flat line.)
+ * 
+ * The anomaly series will have the same hue, but somewhat desaturated colour 
+ * as the original series they represent.They will not appear in legends or 
+ * tooltips, but their name (if needed for further graph manipulation)
+ * will be "[original name] Anomaly."
+ * 
+ * This is intended to display change over time on a single graph.
+ */
+
+var makeAnomalyGraph = function(base, graph) {
+  
+  if(!_.isUndefined(graph.axis.y2)) {
+    throw new Error("Error: Cannot calculate anomalies for multiple data types.");
+  }
+  
+  const baseSeries = _.find(graph.data.columns, series => {return series[0] === base});
+  if(_.isUndefined(baseSeries)) {
+    throw new Error("Error: Invalid base data for anomaly calculation.");
+  }
+  
+  const origLength = graph.data.columns.length;
+  graph.data.axes = {};
+  graph.axis.y2 = {show: true};
+  
+  for(let i = 0; i < origLength; i++) {
+    if(graph.data.columns[i][0] !== 'x') {
+      let oldSeries = graph.data.columns[i];
+      if(oldSeries.length !== baseSeries.length) {
+        throw new Error("Error: Incorrect data series length, cannot calculate anomaly");
+      }
+
+      let newSeries = [];
+      newSeries.push(`${oldSeries[0]} Anomaly`);
+      for(let j = 1; j < oldSeries.length; j++){
+        newSeries.push(oldSeries[j] - baseSeries[j]);
+      }
+      graph.data.columns.push(newSeries);
+      graph.data.axes[oldSeries[0]] = 'y';
+      graph.data.axes[`${oldSeries[0]} Anomaly`] = 'y2';
+      graph.data.types[`${oldSeries[0]} Anomaly`] = oldSeries[0] === base ? "line" : "bar";
+    }
+  }
+  graph.axis.y2.label = {};
+  graph.axis.y2.label.position = 'outer-middle';
+  graph.axis.y2.label.text = `${getAxisTextForVariable(graph, baseSeries[0])} anomaly from ${baseSeries[0]}`;
+  
+  
+  // function that determines whether a data series an anomaly series.
+  // used to format anomalies differently than data
+  let isAnomaly = function (series) {
```

Comment:
Please use just a function declaration: `function isAnomaly(series) { ... }`.

Also, `const` here would be the appropriate declaration; `let` is for values that will be reassigned, i.e., that are variable and that does not seem to be your intention here.